lib/IDE/ExprContextAnalysis.cpp

```diff
@@ -683,6 +679,18 @@ static bool collectPossibleCalleesForUnresolvedMember(
                                               member.Decl))
         candidates.push_back(member);
     }
+  };
+
+  // Get the context of the expression itself.
+  ExprContextInfo contextInfo(&DC, unresolvedMemberExpr);
+  for (auto expectedTy : contextInfo.getPossibleTypes()) {
+    if (!expectedTy->mayHaveMembers())
+      continue;
+    collectMembers(expectedTy);
+    // If this is an optional type, let's also check the base type.
+    if (auto baseOptionalTy = expectedTy->getOptionalObjectType()) {
+      collectMembers(baseOptionalTy);
```

Comment:
```suggestion
    if (auto baseOptionalTy = expectedTy->getOptionalObjectType()) {
      collectMembers(baseOptionalTy->lookThroughAllOptionalTypes());
```
Just like
https://github.com/apple/swift/blob/master/lib/IDE/CodeCompletion.cpp#L4312-L4315

Reply:
Done!